lib/services/local/usagereporter.go

```diff
@@ -0,0 +1,544 @@
+			Event: &prehogapi.SubmitEventRequest_UiOnboardAddFirstResourceClick{
+				UiOnboardAddFirstResourceClick: (*prehogapi.UIOnboardAddFirstResourceClickEvent)(e),
+			},
+		}, nil
+	case *services.UsageUIOnboardAddFirstResourceLaterClickEvent:
+		return &prehogapi.SubmitEventRequest{
+			ClusterName: clusterName,
+			Timestamp:   time,
+			Event: &prehogapi.SubmitEventRequest_UiOnboardAddFirstResourceLaterClick{
+				UiOnboardAddFirstResourceLaterClick: (*prehogapi.UIOnboardAddFirstResourceLaterClickEvent)(e),
+			},
+		}, nil
+	case *services.UsageUIOnboardSetCredentialSubmit:
+		return &prehogapi.SubmitEventRequest{
+			ClusterName: clusterName,
+			Timestamp:   time,
+			Event: &prehogapi.SubmitEventRequest_UiOnboardSetCredentialSubmit{
+				UiOnboardSetCredentialSubmit: (*prehogapi.UIOnboardSetCredentialSubmitEvent)(e),
+			},
+		}, nil
+	case *services.UsageUIOnboardRegisterChallengeSubmit:
+		return &prehogapi.SubmitEventRequest{
+			ClusterName: clusterName,
+			Timestamp:   time,
+			Event: &prehogapi.SubmitEventRequest_UiOnboardRegisterChallengeSubmit{
+				UiOnboardRegisterChallengeSubmit: (*prehogapi.UIOnboardRegisterChallengeSubmitEvent)(e),
+			},
+		}, nil
+	case *services.UsageUIOnboardRecoveryCodesContinueClick:
+		return &prehogapi.SubmitEventRequest{
+			ClusterName: clusterName,
+			Timestamp:   time,
+			Event: &prehogapi.SubmitEventRequest_UiOnboardRecoveryCodesContinueClick{
+				UiOnboardRecoveryCodesContinueClick: (*prehogapi.UIOnboardRecoveryCodesContinueClickEvent)(e),
+			},
+		}, nil
+	default:
+		return nil, trace.BadParameter("unexpected event usage type %T", event)
+	}
+}
+
+func (r *UsageReporter) SubmitAnonymizedUsageEvents(events ...services.UsageAnonymizable) error {
+	var anonymized []*prehogapi.SubmitEventRequest
+
+	for _, e := range events {
+		e.Anonymize(r.anonymizer)
+
+		converted, err := r.convertEvent(e)
+		if err != nil {
+			return trace.Wrap(err)
+		}
+
+		anonymized = append(anonymized, converted)
+
+		usageEventsSubmitted.Inc()
+	}
+
+	r.events <- anonymized
+
+	return nil
+}
+
+// resubmitEvents resubmits events that have already been processed (in case of
+// some error during submission).
+func (r *UsageReporter) resubmitEvents(events []*prehogapi.SubmitEventRequest) {
+	usageEventsRequeuedTotal.Add(float64(len(events)))
+
+	r.events <- events
+}
+
+func NewPrehogSubmitter(ctx context.Context, prehogEndpoint string, clientCert *tls.Certificate, caCertPEM []byte) (UsageSubmitFunc, error) {
+	tlsConfig := &tls.Config{
+		// Self-signed test licenses may not have a proper issuer and won't be
+		// used if just passed in via Certificates, so we'll use this to
+		// explicitly set the client cert we want to use.
+		GetClientCertificate: func(cri *tls.CertificateRequestInfo) (*tls.Certificate, error) {
+			return clientCert, nil
+		},
+	}
+
+	if caCertPEM != nil {
+		pool := x509.NewCertPool()
+		pool.AppendCertsFromPEM(caCertPEM)
+
+		tlsConfig.RootCAs = pool
+	}
+
+	httpClient := &http.Client{
```

Comment:
Use `defaults.HTTPClient()` instead? (and then make any additional customizations)

Reply:
done, though tbh it seems quite a bit uglier than just creating it from scratch, ah well